homeassistant/components/fronius/coordinator.py

```diff
@@ -40,7 +42,7 @@ class FroniusCoordinatorBase(
 
     default_interval: timedelta
     error_interval: timedelta
-    valid_descriptions: list[FroniusSensorEntityDescription]
+    valid_descriptions: dict[Platform, list[FroniusEntityDescription]]
```

Comment:
```suggestion
    valid_descriptions: Mapping[Platform, list[FroniusEntityDescription]]
```
maybe use `Mapping` as type annotation to mark that read-only / non-mutable.

Reply:
Hmm. This causes issues because mypy won't let you call .copy() on a Mapping, I think.

Reply:
```py
from collections.abc import Mapping
from copy import copy


x: Mapping[str, int]
y = copy(x)
``` 
seems to work fine according to https://mypy-play.net/?mypy=latest&python=3.12&flags=strict

Reply:
(But yeah, mypy-play.net says that is fine. Neat tool, I've never seen that before).

Reply:
Oh 🤔 that's unfortunate that it behaves differently locally. 
Are you sure you are running latest mypy locally since I can't seem to be able to reproduce that. Also the stub for `deepcopy` looks like
```py
_T = TypeVar("_T")
def deepcopy(x: _T, memo: dict[int, Any] | None = None, _nil: Any = []) -> _T: ...
```
which doesn't seem to care at all about the type of the first argument.

A workaround would be to use `deepcopy(dict(valid_descriptions))` 🙃